src/Raven.Client/Json/Converters/BlittableJsonConverter.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Sparrow.Json;
+
+namespace Raven.Client.Json.Converters
+{
+
+    internal sealed class BlittableJsonConverter : RavenJsonConverter
+    {
+        public static readonly BlittableJsonConverter Instance = new BlittableJsonConverter();
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            writer.WriteValue(value);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (!(reader is BlittableJsonReader blittableReader))
+            {
+                throw new SerializationException($"{nameof(BlittableJsonReader)} must to be used for convert to {nameof(BlittableJsonReaderObject)}");
+            }
+
+            if (blittableReader.Value == null)
+            {
+                return null;
+            }
+
+            if (blittableReader.Value is BlittableJsonReaderObject blittableValue)
+            {
+                if (reader.TokenType == JsonToken.StartObject)
+                {
+                    //Skip in order to prevent unnecessary movement inside the blittable
+                    //when field\property type is BlittableJsonReaderObject
+                    blittableReader.SkipBlittableInside();
+                }
+
+                return
+                    blittableValue.BelongsToContext(blittableReader.Context) &&
+                    blittableValue.HasParent == false
+                        ? blittableValue
+                        : blittableValue.Clone(blittableReader.Context);
+            }
+            throw new SerializationException($"Try to read {nameof(BlittableJsonReaderObject)} from non {nameof(BlittableJsonReaderObject)} value");
```

Comment:
What _is_ the value that you go? We need to provide as many details as possible to make it easy for the user to understand.

Reply:
done